htdocs/core/tpl/objectline_create.tpl.php

```diff
@@ -176,14 +176,21 @@
 		$filtertype='';
 		if (! empty($object->element) && $object->element == 'contrat' && empty($conf->global->CONTRACT_SUPPORT_PRODUCTS)) $filtertype='1';
 
+		//type must be changed after duration because the change() is hooked on seltype
+		echo '<input type="hidden" id="duration_unit" name="duration_unit" value="">';
```

Comment:
The duration_unit form is filled from https://github.com/Dolibarr/dolibarr/pull/4625/files#diff-6e68d68de845b9bbdcedd79d14c4c5caR193 via ajax, so when the user submits the form it sends the unit to POST for later use